deblur/workflow.py

```diff
@@ -783,29 +790,33 @@ def launch_workflow(seqs_fp, working_dir, mean_error, error_dist,
                      output_fp=output_derep_fp,
                      min_size=min_size, threads=threads_per_sample)
     # Step 3: Remove artifacts
-    output_artif_fp = remove_artifacts_seqs(seqs_fp=output_derep_fp,
-                                            ref_fp=ref_fp,
-                                            working_dir=working_dir,
-                                            ref_db_fp=ref_db_fp,
-                                            negate=True,
-                                            threads=threads_per_sample,
-                                            sim_thresh=sim_thresh)
+    output_artif_fp, num_seqs_left = remove_artifacts_seqs(seqs_fp=output_derep_fp,
+                                                           ref_fp=ref_fp,
+                                                           working_dir=working_dir,
+                                                           ref_db_fp=ref_db_fp,
+                                                           negate=True,
+                                                           threads=threads_per_sample,
+                                                           sim_thresh=sim_thresh)
     if not output_artif_fp:
         warnings.warn('Problem removing artifacts from file %s' %
                       seqs_fp, UserWarning)
         logger.warning('remove artifacts failed, aborting')
         return None
     # Step 4: Multiple sequence alignment
-    output_msa_fp = join(working_dir,
-                         "%s.msa" % basename(output_artif_fp))
-    alignment = multiple_sequence_alignment(seqs_fp=output_artif_fp,
-                                            threads=threads_per_sample)
-    if not alignment:
-        warnings.warn('Problem performing multiple sequence alignment '
-                      'on file %s' % seqs_fp, UserWarning)
-        logger.warning('msa failed. aborting')
-        return None
-
+    if num_seqs_left > 1:
+        output_msa_fp = join(working_dir,
+                             "%s.msa" % basename(output_artif_fp))
+        alignment = multiple_sequence_alignment(seqs_fp=output_artif_fp,
+                                                threads=threads_per_sample)
+        if not alignment:
+            warnings.warn('Problem performing multiple sequence alignment '
+                          'on file %s' % seqs_fp, UserWarning)
+            logger.warning('msa failed. aborting')
+            return None
+    else:
+        # only one sequence after remove artifacts (but could be many reads)
```

Comment:
the pipeline should take care of it by failing the next step
but just to be on the safe/clear side, added explicitly num_seqs_left == 0 condition